dump/dump.c

```diff
@@ -64,26 +64,34 @@ static unsigned int exfat_count_used_clusters(unsigned char *bitmap,
 	return count;
 }
 
+
+static off_t c2o(off_t heap_offset, unsigned int clu, unsigned int clu_size)
+{
+	return heap_offset + (clu - EXFAT_FIRST_CLUSTER) * clu_size;
+}
+
 static int exfat_show_ondisk_all_info(struct exfat_blk_dev *bd)
 {
 	struct pbr *ppbr;
 	struct bsx64 *pbsx;
 	struct exfat_dentry *ed;
+	unsigned int sect_size, clu_size;
 	unsigned int root_clu_off, bitmap_clu_off, bitmap_clu;
-	unsigned int total_clus, used_clus, clu_offset, root_clu;
+	unsigned int total_clus, used_clus, root_clu;
+	off_t heap_offset;
 	unsigned long long bitmap_len;
 	int ret;
 	unsigned char *bitmap;
 	char *volume_label;
 
-	ppbr = malloc(bd->sector_size);
+	ppbr = malloc(sizeof(*ppbr));
```

Comment:
As I know, we should access  sector as per 4k unit in case of native 4K sector.(i.e. hw sector size : 4K, logical sector size : 4K)

Reply:
I will declare EXFAT_MAX_SECTOR_SIZE 4096 macro, we can use it here.

Reply:
Okay, I will change this.

Reply:
Ah, Reading small data with read(2) can be a problem? read(2) is buffered I/O. Let me check this.

Reply:
Do you have 4K native device ? you should read under 4KB against block device, not filesystem.

Reply:
We read data from a block device file which is one of file in "bdev" filesystem. Anyway I will apply this and the above comments